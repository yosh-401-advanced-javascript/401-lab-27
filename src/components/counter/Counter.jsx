import React from 'react';


class Counter extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      counter: 0,
      error: 'error'
    };
  }

  /**
   * @Function to handle increments
   * @param e
   */
  handleIncrement = e => {
    // e.preventDefault();
    this.setState((previousState) => {
      return { counter: previousState.counter + 1 };
    })
  };

  /**
   * @Function to handle decrements
   * @Function changes the color of the counter
   * @param e
   */
  handleDecrement = e => {
    // e.preventDefault();
    this.setState((previousState) => {
      if(this.state.counter <= 0){
        return { error: 'error'};
      } else {
        return { counter: previousState.counter - 1, error: 'counter' };
      }
    })
  };

  render() {
    return (
        <div className='counter'>
          <h4 className={this.state.error}>{this.state.counter}</h4>
          <button className='incrementButton' onClick={this.handleIncrement}>Increment me!
          </button>
          <button className='decrementButton' onClick={this.handleDecrement} >Decrement me!
          </button>
        </div>
    );
  }
}

export default Counter
