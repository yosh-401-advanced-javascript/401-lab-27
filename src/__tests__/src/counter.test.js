import React from 'react';
import renderer from 'react-test-renderer';
import Counter from '../../components/counter/Counter'
import Footer from '../../components/footer/footer'
import Header from '../../components/header/header'
import App from '../../app'

describe('Testing Counter', () => {
  it('is alive at application start', function() {
    let app = shallow(<Counter />);
    expect(app.find('h4').exists()).toBeTruthy();
  });
});
// it('should change state on click', () => {
//   let app = mount(<Counter />);
//   let button = app.find('button');
//   button.simulate('click');
//   expect(app.state('counter')).toBe(0);
//   expect(app.find('section').text()).toContain('')
//
//
// });

it('should render correctly', () => {
  const tree = renderer.create(<Counter />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should increment the count', () => {
  let app = shallow(<Counter />);
  let incrementButton = app.find('.incrementButton');
  incrementButton.simulate('click');
  let text = app.find("h4").text();
  expect(text).toEqual('1');
});
it('should decrement the count', () => {
  let app = shallow(<Counter />);
  let decrementButton = app.find('.decrementButton');
  decrementButton.simulate('click')
  let text = app.find('h4').text();
  expect(text).toEqual('0');
});

describe('Testing Footer ', () => {
  it('should should render correctly', () => {
    const tree = renderer.create(<Footer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});


describe('Testing Header ', () => {
  it('should should render correctly', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();

  });

});
describe('Testing App ', () => {
  it('should should render correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();

  });

});

