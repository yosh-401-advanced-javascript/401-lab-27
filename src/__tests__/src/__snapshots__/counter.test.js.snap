// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Footer  should should render correctly 1`] = `
<footer>
  Â© 2018 Code Fellows
</footer>
`;

exports[`Testing Header  should should render correctly 1`] = `
<header>
  <h1>
    Header
  </h1>
</header>
`;

exports[`should render correctly 1`] = `
<div
  className="counter"
>
  <h4
    className="error"
  >
    0
  </h4>
  <button
    className="incrementButton"
    onClick={[Function]}
  >
    Increment me!
  </button>
  <button
    className="decrementButton"
    onClick={[Function]}
  >
    Decrement me!
  </button>
</div>
`;
